kvbc/src/categorization/block_merkle_category.cpp

```diff
@@ -511,6 +555,43 @@ void BlockMerkleCategory::deleteGenesisBlock(BlockId block_id, const BlockMerkle
   deleteStaleData(out.state_root_version, batch);
 }
 
+void BlockMerkleCategory::deleteLastReachableBlock(BlockId block_id,
+                                                   const BlockMerkleOutput& out,
+                                                   NativeWriteBatch& batch) {
+  for (const auto& [key, _] : out.keys) {
+    (void)_;
+    const auto hashed_key = KeyHash{hash(key)};
+    const auto versioned_key = serializeThreadLocal(VersionedKey{hashed_key, block_id});
+    // Find the previous version of the key and set it as a last version. Exploit the fact that CMF
+    // serializes strings prefixed by their length (in big-endian). Therefore, VersionedKeys will be ordered by key
+    // size, key and version. If not found, then this is the only version of the key and we can remove the latest
+    // version index too.
```

Comment:
The `Exploit the fact that CMF serializes strings prefixed by their length (in big-endian)` and ordered by `key size` parts can be removed for the hashed keys that are `fixedlist`s.

Reply:
Fixed.